src/components/Aside/index.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Link as GatsbyLink } from 'gatsby';
+import { Markdown } from '../index';
+
+const Link = ({ to, activeStyle, name }) => {
+  const internal = /^\/(?!\/)/.test(to);
+  if (internal) {
+    return (
+      <GatsbyLink to={to} activeStyle={activeStyle}>
+        <Markdown>{name}</Markdown>
+      </GatsbyLink>
+    );
+  }
+  return (
+    <a href={to}>
+      <Markdown>{name}</Markdown>
+    </a>
+  );
+};
+
+Link.propTypes = {
+  to: PropTypes.string,
+  activeStyle: PropTypes.object,
+  name: PropTypes.string,
+};
+
+class Aside extends React.Component {
+  render() {
+    const asideData = [];
+    const data = this.props.children.split('\n').filter(el => el !== '');
+    const findLinks = /\(([^)]+)\)/;
+    const findNames = /\[(.*?)\]/;
+    data.map(el => {
```

Comment:
You might want to consider instead providing a custom component for the <a> tag to the Markdown component which is smart about doing the GatsbyLink versus an Anchor or Link based on the href. I considered doing this in general to all the usages of Markdown. I also wanted to make it smart about links to github, twitter, etc and add the corresponding icon for those automatically rather than trying to embed that image in the markdown where needed.